### PR TITLE
[MAINTENANCE] Add PR title checker GitHub Action

### PR DESCRIPTION
**Pull request by cdkini**
Changes proposed in this pull request:
- Add check to ensure consistent names

### Definition of Done
- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code

### CODE REVIEW

File: .github/workflows/pr-title-checker.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Validate that PR titles are prefixed with one of our accepted labels
+name: "PR Title Checker"
+on:
+  pull_request:
+    types:
+      - opened
+      - edited
+      - reopened
+      - synchronize
+      - auto_merge_enabled
```
Comment (NathanFarmer) [inline .github/workflows/pr-title-checker.yml:10]: Is there a list of possibilities somewhere?

**Review by NathanFarmer — APPROVED** [review]
One non-blocking question

**Review by donaldheppner — APPROVED** [review]
thank you for this!

File: .github/workflows/pr-title-checker.yml (hunk 1 above)
Comment (cdkini) [reply .github/workflows/pr-title-checker.yml:10]: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request

**Pull request merged**